object/init_data.go

```diff
@@ -106,7 +106,19 @@ func initDefinedApplication(application *Application) {
 		return
 	}
 	application.CreatedTime = util.GetCurrentTime()
-	AddApplication(application)
+	// If user not defined clientId and clientSecret
+	if len(application.ClientId) == 0 || len(application.ClientSecret) == 0 {
+		application.ClientId = util.GenerateClientId()
+		application.ClientSecret = util.GenerateClientSecret()
+	}
+	for _, providerItem := range application.Providers {
+		providerItem.Provider = nil
+	}
+
+	_, err := adapter.Engine.Insert(application)
```

Comment:
Why not use `AddApplication(application)`?

Reply:
`AddApplication` only supports randomly generated clientId and clientSecret.